File: tests/benchmarks/resolution/fixtures/dynamic-swift/dispatch.swift
// Phase 6: Swift dynamic dispatch patterns
// NSSelectorFromString("name") → reflection kind
// performSelector → flagged as unresolved-dynamic
import Foundation

func greet(_ name: String) -> String {
    return "Hello, \(name)"
}

func farewell(_ name: String) -> String {
    return "Goodbye, \(name)"
}

// NSSelectorFromString("greet") — reflection kind, resolves to top-level greet()
func runNSSelectorFromString() -> Selector {
    return NSSelectorFromString("greet")
}

// NSSelectorFromString with farewell
func runNSSelectorFarewell() -> Selector {
    return NSSelectorFromString("farewell")
}

File: tests/benchmarks/resolution/fixtures/dynamic-php/dispatch.php
<?php
// Fixture: PHP dynamic dispatch patterns
// call_user_func('fn', args) → resolved as reflection kind
// $obj->$m() → flagged as unresolved-dynamic
// $fn() → flagged as unresolved-dynamic

function greet(string $name): string {
    return "Hello, {$name}";
}

function farewell(string $name): string {
    return "Goodbye, {$name}";
}

// call_user_func('greet', ...) — resolved to top-level greet()
function runCallUserFuncLiteral(): string {
    return call_user_func('greet', 'world');
}

// $obj->$m() — variable method name; unresolved-dynamic
function runVariableMethod(object $obj, string $m): mixed {
    return $obj->$m('world');
}

// $fn() — variable callable; unresolved-dynamic
function runVariableCallable(callable $fn): mixed {
    return $fn('world');
}
